api/pkg/gateway/context_test.go

```diff
@@ -0,0 +1,541 @@
+
+			require.Equal(t, tc.expected.userName, retGetTenantFst)
+			require.Equal(t, tc.expected.ok, retGetTenantSnd)
+		})
+	}
+}
+
+func TestGetUsername(t *testing.T) {
+	type Expected struct {
+		userName string
+		ok       bool
+	}
+	cases := []struct {
+		description string
+		headers     map[string]string
+		expected    Expected
+	}{
+		{
+			description: "verify if given value returns from header",
+			headers: map[string]string{
+				"X-Username": "someone",
+			},
+			expected: Expected{
+				userName: "someone",
+				ok:       true,
+			},
+		}, {
+			description: "validate empty ID string, for ID behaviour",
+			headers: map[string]string{
+				"X-Username": "",
+			},
+			expected: Expected{
+				userName: "",
+				ok:       false,
+			},
+		}, {
+			description: "validate empty headers, for fail function",
+			headers: map[string]string{
+				"": "",
+			},
+			expected: Expected{
+				userName: "",
+				ok:       false,
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.description, func(t *testing.T) {
+			e := echo.New()
+			req := httptest.NewRequest(http.MethodPost, "/", nil)
+
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+			for k, v := range tc.headers {
+				req.Header.Set(k, v)
+			}
+
+			rec := httptest.NewRecorder()
+			ctxEchoT := e.NewContext(req, rec)
+
+			gatewayCtxT := Context{
+				nil,
+				ctxEchoT,
+			}
+
+			retUserName, retBoolValidate := gatewayCtxT.GetUsername()
+
+			require.Equal(t, tc.expected.userName, retUserName)
+			require.Equal(t, tc.expected.ok, retBoolValidate)
+		})
+	}
+}
+
+func TestGetRole(t *testing.T) {
+	type Expected struct {
+		role string
+		ok   bool
+	}
+	cases := []struct {
+		description string
+		headers     map[string]string
+		expected    Expected
+	}{
+		{
+			description: "verify if given value returns from header",
+			headers: map[string]string{
+				"X-Role": "someone",
+			},
+			expected: Expected{
+				role: "someone",
+				ok:   true,
+			},
+		}, {
+			description: "validate empty ID string, for ID behaviour",
+			headers: map[string]string{
+				"X-Role": "",
+			},
+			expected: Expected{
+				role: "",
+				ok:   false,
+			},
+		}, {
+			description: "validate empty headers, for fail function",
+			headers: map[string]string{
+				"": "",
+			},
+			expected: Expected{
+				role: "",
+				ok:   false,
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.description, func(t *testing.T) {
+			e := echo.New()
+			req := httptest.NewRequest(http.MethodPost, "/", nil)
+
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+			for k, v := range tc.headers {
+				req.Header.Set(k, v)
+			}
+
+			rec := httptest.NewRecorder()
+			ctxEchoT := e.NewContext(req, rec)
+
+			gatewayCtxT := Context{
+				nil,
+				ctxEchoT,
+			}
+
+			retStringRole, retBoolRoleValidate := gatewayCtxT.GetRole()
+
+			require.Equal(t, tc.expected.role, retStringRole)
+			require.Equal(t, tc.expected.ok, retBoolRoleValidate)
```

Comment:
You can compare the entire struct instead of its individual attributes. Change this to:
```go
require.Equal(t, tc.expected, Expected{retStringRole, retBoolRoleValidate})
```